### PR TITLE
[RHCLOUD-19948] feature: validate "edit source" requests

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Makes sure that no source can be edited with an availability status set to empty or null.

## Links

[[RHCLOUD-19948]](https://issues.redhat.com/browse/RHCLOUD-19948)

### CODE REVIEW

File: service/source_validation.go
Comment (petracihalova) [inline service/source_validation.go]: I know that this PR is not ready for review but I found out that you solve here the same name validation as was solved here https://github.com/RedHatInsights/sources-api-go/pull/507 (already merged) but the name of function is slightly different so I wanted to let you know
+
for edit request I believe the "Name" field can be nil (when you dont want to edit the name) so the logic should be updated here ... we need to cover situation when the "name" field is not part of body request and when the "name" field is there and is an empty string 🤔

Comment (MikelAlejoBR) [reply service/source_validation.go]: In this case I think it is better to just run the validations if the incoming name isn't `nil` — if it is nil, we know it wasn't included in the payload, so we don't need to modify it. If it is not `nil`, even if it was an accidental empty string, we enforce the validation.

Comment (petracihalova) [reply service/source_validation.go]: yes and only for info: this is solved in @Ellen-Yi-Dong #544 too

Comment (MikelAlejoBR) [reply service/source_validation.go]: Mmmmmmmmmmm... I'm doubting on whether waiting until that PR is merged and then rebasing this one...

**Review by Ellen-Yi-Dong — APPROVED** [review]

**Review by Ellen-Yi-Dong — COMMENTED** [review]
LGTM!

**Review by petracihalova — APPROVED** [review]
LGTM

**Review by lindgrenj6 — APPROVED** [review]

**Comment by lindgrenj6** [comment]
Looks like we got a conflict, will merge once it's solved!

**Comment by MikelAlejoBR** [comment]
All right everything's green!

**Pull request merged**
